### ISSUE TITLE
Neo.DatabaseError.General.UnknownError  : Expected a sorted plan  related to ORDER BY

### ISSUE THREAD

**Issue by tomzeppenfeldt** [neo4j/neo4j#12784]
Neo4j 4.3.*

A query ending with 

```
RETURN pattern,
       patternText,
       patternLength,
       countEndPoints,
       count ORDER BY patternLength,count
```

triggers the error.

![image](https://user-images.githubusercontent.com/3007812/140621958-4de1f7dc-1e19-4562-9aa8-2342f65afbf6.png)

removing one of the ORDER by fields

```
RETURN pattern,
       patternText,
       patternLength,
       countEndPoints,
       count ORDER BY patternLength
```

returns the expected result
![image](https://user-images.githubusercontent.com/3007812/140621939-df83e772-d84e-4acf-84fd-410c1777ec27.png)

**Comment by mnd999** [comment]
Thanks for the report , we'll investigate and get back to you.

**Comment by arnefischereit** [comment]
Thanks, @tomzeppenfeldt, for handing this in.

To make it easier for us to investigate, could you also provide us with the rest of the query, which we apparently could not sort?

**Comment by tomzeppenfeldt** [comment]
this is the complete query. You should be able to run it on any store, no specific labels are required.

I have the issue mainly on Aura

It takes these params. ids => array of node ids   length => for the apoc.path.expand

```
{
  "length": 3,
  "ids": [
    316211
  ]
}
```

error occurs also when `ORDER BY patternLength DESC`

```
MATCH (n) WHERE id(n) in $ids
WITH n
CALL apoc.path.expand(n,null,null,1,$length) YIELD path

WITH path
WHERE NOT apoc.coll.containsDuplicates(nodes(path))

WITH apoc.path.elements(path) AS pathElements

WITH REDUCE(pattern='', el IN [e IN pathElements | 
            CASE WHEN apoc.coll.indexOf(pathElements,e)%2 = 0 THEN 
               REDUCE(s='(', l in labels(e) |
                       s+':'+l
                )     
               +')'
            ELSE
               CASE WHEN pathElements[apoc.coll.indexOf(pathElements,e)-1] = startNode(e) THEN
                    '-[:'+type(e)+']->'
               ELSE
                    '<-[:'+type(e)+']-'
               END
            END
       ] |
       pattern+el
      ) as pattern, 
      REDUCE(pattern='', el IN [e IN pathElements | 
            CASE WHEN apoc.coll.indexOf(pathElements,e)%2 = 0 THEN 
               REDUCE(s='(', l in labels(e) |
                       s
		      +CASE WHEN s='(' THEN '' ELSE ',' END
		      +l
                )     
               +')'
            ELSE
               CASE WHEN pathElements[apoc.coll.indexOf(pathElements,e)-1] = startNode(e) THEN
                    '-['+type(e)+']->'
               ELSE
                    '<-['+type(e)+']-'
               END
            END
       ] |
       pattern+el
      ) as patternText,
	  toInteger(size(pathElements)/2) AS patternLength,
	  SIZE(collect(DISTINCT LAST(pathElements))) AS countEndPoints, 
	  count(*) AS count
RETURN pattern,patternText,patternLength,countEndPoints,count ORDER BY patternLength, count
```

**Comment by arnefischereit** [comment]
Thanks @tomzeppenfeldt!

We could reproduce the issue and will prioritize it in our upcoming work.

**Comment by arnefischereit** [comment]
Hi @tomzeppenfeldt,
Thank you again for reporting this issue. We [have merged a fix](https://github.com/neo4j/neo4j/commit/ba8d72fbc67095b37a9f417bbc6dd06064a27d52) for this bug and made it publicly available [as part of version 4.3.8](https://github.com/neo4j/neo4j/wiki/Neo4j-4.3-changelog#cypher-2).
